backend/github.js

```diff
@@ -72,4 +72,85 @@ exports.hasAdminAccess = function (repository, accessToken, callback) {
     var adminPermission = bodyJSON.permissions.admin || false;
     callback(null, adminPermission);
   });
-};
\ No newline at end of file
+};
+/**
+* hookValidator is used to check wheather hook is registered to
+* repository or not
+* @param repositoryName: name of the repository
+* @param token: github access token
+*/
+exports.hookValidator = function (repositoryName, token) {
+  return new Promise(function(resolve, reject) {
+    token = crypter.decrypt(token);
+    request({
+      url: `https://api.github.com/repos/${repositoryName}/hooks?access_token=${token}`,
+      headers: {
+        'User-Agent': 'request'
+      }
+    }, function (error, response, body) {
+      var isRegistered = false;
+      if (response.statusCode !== 200) {
+        console.log(response.statusCode + ': ' + response.statusMessage);
+        reject()
+      }
+      var hooks = JSON.parse(body);
+      var hookurl = 'https://' + process.env.HOSTNAME + '/ci/webhook';
```

Comment:
do not strict it to https if a deployment does not have SSL encryption the webhooks will fail. Had this problem in my deployment